ui/e2e/src/pages/DashboardPage.ts

```diff
@@ -230,4 +242,28 @@ export class DashboardPage {
   getVariableEditor() {
     return new VariableEditor(this.variableEditor);
   }
+
+  /**
+   * MOCKING NETWORK REQUESTS
+   */
+  async mockQueryRangeRequests({ queries }: MockQueryRangeConfig) {
+    // Mock data response, so we can make assertions on consistent response data.
+    await this.page.route('**/api/v1/query_range', (route) => {
+      const request = route.request();
+      const requestPostData = request.postDataJSON();
+
+      const requestQuery = typeof requestPostData === 'object' ? requestPostData['query'] : undefined;
+      const mockQuery = queries.find((mockQueryConfig) => mockQueryConfig.query === requestQuery);
+
+      if (mockQuery) {
+        route.fulfill(mockQuery.response);
+      } else {
+        route.continue();
+      }
+    });
+  }
+
+  async cleanupMockRequests() {
```

Comment:
Intentionally baking this in and calling it in the fixture to avoid having to remember to always call it in `beforeEach`. If you forget the cleanup, the browser behaves weirdly when running headed tests using vscode, and I _think_ it's possible they'd mess up queries in other tests, but I haven't tested it.